File: Mock_2/countingstars/CountingStars.java
import java.io.FileNotFoundException;
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class CountingStars {

    static boolean[][] visited;
    public static void main(String[] args) throws FileNotFoundException {
        Scanner sc = new Scanner(System.in);

        int caseNum = 0;
        int stars;
        while (sc.hasNext()) {
            caseNum++;
            stars = 0;
            int m = sc.nextInt();
            int n = sc.nextInt();
            visited = new boolean[m][n];
            sc.nextLine();
            List<String> sky = new ArrayList<>();
            for (int i = 0; i < m; i++) {
                sky.add(sc.nextLine());
            }

            for (int row = 0; row < m; row++) {
                for (int col = 0; col < n; col++) {
                    if (! visited[row][col] && sky.get(row).charAt(col) == '-') {
                        stars++;
                        markVisited(sky, row, col);
                    }
                }
            }
            System.out.println("Case " + caseNum + ": " + stars);
        }
        sc.close();
    }
    public static int[][] getNeighbors(int row, int col) {
        return new int[][]{
            {row-1, col}, {row, col-1}, {row, col+1}, {row+1, col}
        };
    }
    public static boolean inBounds(List<String> sky, int row, int col) {
        return 0 <= row && row < sky.size() && 0 <= col && col < sky.get(row).length();
    }
    public static void markVisited(List<String> sky, int row, int col) {
        if (!inBounds(sky, row, col) || visited[row][col] || sky.get(row).charAt(col) != '-')
            return;
        
        visited[row][col] = true;
        for (int[] pos : getNeighbors(row, col)) {
            markVisited(sky, pos[0], pos[1]);
        }
    }
}

File: Mock_2/countingstars/countingstars.py
from sys import stdin, setrecursionlimit
setrecursionlimit(5000)

lines = [line.strip() for line in stdin]

index = 0
visited = set()


def get_neighbors(x, y):
    return (x, y-1), (x-1, y), (x+1, y), (x, y+1)
def in_bounds(sky, x, y):
    return 0 <= x < len(sky) and 0 <= y < len(sky[x])
def dfs(sky, x, y):
        if not in_bounds(sky, x, y) or (x, y) in visited or sky[x][y] != '-':
            return
        visited.add((x, y))
        for neighbor in get_neighbors(x, y):
            dfs(sky, neighbor[0], neighbor[1])

case_num = 1
while index < len(lines):
    visited = set()  
    m, n = map(int, lines[index].split())
    sky = [lines[i] for i in range(index+1, index+m+1)]
    
   
    stars = 0
    for x in range(len(sky)):
        for y in range(len(sky[0])):
                if in_bounds(sky, x, y) and (x, y) not in visited:
                    a = sky[x][y]
                    if a == '-':
                        stars += 1
                        dfs(sky, x, y)

    print(f'Case {case_num}: {stars}')

    index += m + 1
    case_num += 1
